src/Codegen/Escaper.php

```diff
@@ -0,0 +1,31 @@
+<?php declare(strict_types=1);
+
+namespace Spawnia\Sailor\Codegen;
+
+use function explode;
+use function implode;
+use Nette\PhpGenerator\Helpers;
+use function strtolower;
+
+class Escaper
+{
+    public static function escapeName(string $name): string
+    {
+        // Inspiration from https://github.com/nette/php-generator/blob/f19b7975c7c4d729be5b64fce7eb72f0d4aac6fc/src/PhpGenerator/ClassLike.php#L87
+        if (isset(Helpers::Keywords[strtolower($name)])) {
+            $name = '_' . $name;
+        }
+
+        return $name;
+    }
+
+    public static function escapeNamespace(string $name): string
+    {
+        $parts = explode('\\', $name);
+        foreach ($parts as $i => $part) {
+            $parts[$i] = self::escapeName($part);
```

Comment:
```suggestion
        foreach ($parts as &$part) {
            $part = self::escapeName($part);
```

Reply:
TBH not sure what's better.

I know that with using references there's risk of overwriting the variable later since it's not scoped so I prefer to use indices.

Reply:
A quick benchmark confirmed my intuition that my suggested approach is faster:

![image](https://user-images.githubusercontent.com/12158000/177575345-2fcc8278-325f-4465-9630-478dc6d8ecf8.png)